src/pages/collators/CollatorRewards.tsx

```diff
@@ -0,0 +1,160 @@
+    };
+
+    fetchAvailableBalance().then((balance) => setUserAvailableBalance(balance));
+  }, [api, walletAccount]);
+
+  const updateRewardsExtrinsic = useMemo(() => {
+    if (!api) {
+      return undefined;
+    }
+
+    return createUpdateDelegatorRewardsExtrinsic();
+  }, [api, createUpdateDelegatorRewardsExtrinsic]);
+
+  const submitUpdateExtrinsic = useCallback(() => {
+    if (!updateRewardsExtrinsic || !api || !walletAccount) {
+      return;
+    }
+    setSubmissionPending(true);
+    updateRewardsExtrinsic
+      .signAndSend(walletAccount.address, { signer: walletAccount.signer as any }, (result) => {
+        const { status, events } = result;
+        const errors = getErrors(events, api);
+        if (status.isInBlock) {
+          if (errors.length > 0) {
+            const errorMessage = `Transaction failed with errors: ${errors.join('\n')}`;
+            console.error(errorMessage);
+            toast(errorMessage, { type: 'error' });
+          }
+        } else if (status.isFinalized) {
+          setSubmissionPending(false);
+          if (errors.length === 0) {
+            toast('Delegator rewards updated', { type: 'success' });
+          }
+        }
+      })
+      .catch((error) => {
+        toast('Transaction submission failed: ' + error.toString(), {
+          type: 'error',
+        });
+        setSubmissionPending(false);
+      });
+  }, [api, setSubmissionPending, updateRewardsExtrinsic, walletAccount]);
+
+  return (
+    <>
+      <div className="flex mb-8 justify-between">
+        <div className="card gap-0 rounded-lg bg-base-200 w-1/2 mr-4 collators-box">
+          <div className="card-body">
+            <h2 className="card-title">Collators</h2>
+            <div className="flex flex-row">
+              <div className="flex-initial pr-5">
+                <StakedIcon />
+              </div>
+              <div className="flex-auto">
+                <h3>{nativeToFormat(userStaking?.amount || '0.00', tokenSymbol)}</h3>
+                <p>My Staking</p>
+              </div>
+              <div className="flex-auto">
+                <h3>{nativeToFormat(userAvailableBalance, tokenSymbol)}</h3>
+                <p>Free balance</p>
+              </div>
+              <div className="flex flex-auto place-content-end">
+                <button className="btn btn-secondary w-full" disabled>
+                  0 {tokenSymbol} Unboarding
+                </button>
+              </div>
+            </div>
+          </div>
+        </div>
+        <div className="card rounded-lg bg-base-200 w-1/2 ml-4 collators-box">
+          <div className="card-body">
+            <h2 className="card-title">Staking Rewards</h2>
+            <div className="flex flex-row">
+              <div className="flex-initial pt-1 pr-5 pb-0">
+                <RewardsIcon />
+              </div>
+              <div className="flex-auto">
+                <h4>{nativeToFormat(estimatedRewards, tokenSymbol)}</h4>
```

Comment:
The balance here is not updated after clicking on the 'update' button. Reason being that this is kind of cached in the `useStakingPallet()` hook. We should consider changing the function in that hook such that it's possible to fetch the latest estimate of rewards.

Reply:
Okay, I'll take a look at this too.

Reply:
So okay, what happens is that useEffect() won't change if no dependency changed, and in this case the dependency is "hidden" because the value changes remotely. So I simply added a function to refresh the estimated rewards.